src/intrinsics.cpp

```diff
@@ -918,6 +920,180 @@ static Value *emit_intrinsic(intrinsic f, jl_value_t **args, size_t nargs,
+        ret->addIncoming(here_byte, here);
+        ret->addIncoming(there_byte, there);
+        builder.Insert(ret);
+        return ret;
+    }
+    HANDLE(bytevec_ref32,2) {
+        Value *b = JL_INT(x);
+        Value *i = builder.CreateSub(JL_INT(y), ConstantInt::get(T_size, 1));
+        bool check_bounds =
+#if CHECK_BOUNDS==1
+            ((ctx->boundsCheck.empty() || ctx->boundsCheck.back() == true) &&
+             jl_options.check_bounds != JL_OPTIONS_CHECK_BOUNDS_OFF) ||
+              jl_options.check_bounds == JL_OPTIONS_CHECK_BOUNDS_ON;
+#else
+            0;
+#endif
+        BasicBlock *here  = BasicBlock::Create(getGlobalContext(), "here",  ctx->f);
+        BasicBlock *check = BasicBlock::Create(getGlobalContext(), "check", ctx->f);
+        BasicBlock *there = BasicBlock::Create(getGlobalContext(), "there", ctx->f);
+        BasicBlock *oob   = BasicBlock::Create(getGlobalContext(), "oob",   ctx->f);
+        BasicBlock *cont  = BasicBlock::Create(getGlobalContext(), "cont",  ctx->f);
+
+        // branch: "here" or "there"
+        Value *is_here = builder.CreateICmpSGE(b, ConstantInt::get(b->getType(), 0));
+        Value *here_len = builder.CreateZExt(
+            builder.CreateTrunc(
+                builder.CreateLShr(
+                    b, ConstantInt::get(b->getType(), 8*(2*sizeof(void*)-1))
+                ), T_uint8
+            ), T_size
+        );
+        builder.CreateCondBr(
+            !check_bounds ? is_here :
+            builder.CreateAnd(is_here, builder.CreateICmpULT(i, here_len)),
+            here, check_bounds ? check : there
+        );
+
+        // decode here byte (known to be in-bounds)
+        builder.SetInsertPoint(here);
+        Value *here_uint32 = builder.CreateTrunc(
+            builder.CreateLShr(
+                b,
+                builder.CreateZExt(
+                    builder.CreateShl(i, ConstantInt::get(T_size, 3)),
+                    b->getType()
+                )
+            ),
+            T_uint32
+        );
+        builder.CreateBr(cont);
+
+        // check bounds (here & there)
+        if (check_bounds) {
+            builder.SetInsertPoint(check);
+            Value *there_len = builder.CreateSub(
+                ConstantInt::get(T_size, 0),
+                builder.CreateTrunc(
+                   builder.CreateAShr(b, ConstantInt::get(b->getType(), 8*sizeof(void*))),
+                   T_size
+                )
+            );
+            builder.CreateCondBr(
+                builder.CreateAnd(builder.CreateNot(is_here), builder.CreateICmpULT(i, there_len)),
+                there, oob
+            );
+
+            // raise out-of-bounds error
+            builder.SetInsertPoint(oob);
+            builder.CreateCall2(
+                prepare_call(jlthrow_line_func),
+                tbaa_decorate(tbaa_const, builder.CreateLoad(prepare_global(jlemptysvec_var))),
+                ConstantInt::get(T_int32, ctx->lineno)
```

Comment:
i don't think `throw(Base.svec())` is a particularly legal statement in Julia (although it does seem to work at the repl). there are a lot of bounds errors functions that you can call however (see `emit_bounds_check` in cgutils.cpp)